src/query_parser/v2/parser.y

```diff
@@ -381,77 +410,21 @@ expr(A) ::= union(B) . [ORX] {
 }
 
 union(A) ::= expr(B) OR expr(C) . [OR] {
-    int rv = one_not_null(B, C, (void**)&A);
-    if (rv == NODENN_BOTH_INVALID) {
-        A = NULL;
-    } else if (rv == NODENN_ONE_NULL) {
-        // Nothing- already assigned
-    } else {
-        if (B->type == QN_UNION && B->opts.fieldMask == RS_FIELDMASK_ALL) {
-            A = B;
-        } else {
-            A = NewUnionNode();
-            QueryNode_AddChild(A, B);
-            A->opts.fieldMask |= B->opts.fieldMask;
-        }
-        // Handle C
-        QueryNode_AddChild(A, C);
-        A->opts.fieldMask |= C->opts.fieldMask;
-        QueryNode_SetFieldMask(A, A->opts.fieldMask);
-    }
+  A = union_step(B, C);
```

Comment:
Why not use the same for `v1`?

Reply:
because on V1 there are only 2 places for that and I got lazy. The Lemon detects that it's the same code and doesn't copy it anyways

Reply:
The parser is complicated enough it would be nice to at least have it uniform.
But `v1` should be thrown away hopefully soon so won't insist